### ISSUE TITLE
Supported platform for package confusion in UI

### ISSUE THREAD

**Issue by TheLunaticScripter** [habitat-sh/builder#716]
So not sure if this is only specific to this package or not but the 7zip package shows that the version is built for both windows and linux but when uploaded to the on-prem builder only the windows package is uploaded. In the UI the supported platform shows linux but under the `target_os` tag it only shows windows.

![image](https://user-images.githubusercontent.com/16371030/46688454-62221300-cbcb-11e8-8852-a6fbd5a20693.png)

**Comment by smacfarlane** [comment]
The package in question has a Linux filename: `core-7zip-16.04-20170131110814-x86_64-linux.hart`  but if you extract the file on disk, the `TARGET` metadata says it's a `x86_64-windows` package and contents are Windows files.    I've removed the package from stable so that in most uses you won't encounter this problem. 

@chefsalim   I've attempted to reproduce this issue in acceptance, but haven't been able to so far.  My suspicion is that the scenario that caused this is no longer possible after the PackageTarget work done earlier this summer.
